lib/web_ui/lib/src/engine/skwasm/skwasm_impl/renderer.dart

```diff
@@ -285,9 +295,85 @@ class SkwasmRenderer implements Renderer {
       indices: indices
     );
 
+  ui.Size? _scaledSize(
+    int width,
+    int height,
+    int? targetWidth,
+    int? targetHeight,
+  ) {
+    if (targetWidth == null && targetHeight == null) {
+      return null;
+    }
+    if (targetWidth == width && targetHeight == height) {
+      // Not scaled
+      return null;
+    }
+    if (targetWidth == null) {
+      if (targetHeight == height) {
+        // Not scaled.
+        return null;
+      }
+      targetWidth = (width * targetHeight! / height).round();
```

Comment:
Looks like the compiler failed to figure out that `targetHeight` is not null nere. Or am I missing something? Should we file an issue?

Reply:
I think expecting the compiler to figure out that `targetHeight` is non-null here is a little unrealistic. I might be able to restructure the code a little to make it more obvious to the compiler though... Hmmm...